search/routes/api/serialize.py

```diff
@@ -122,31 +127,84 @@ def as_json(document_or_set: Union[DocumentSet, Document],
 class AtomXMLSerializer(BaseSerializer):
     """Atom XML serializer for paper metadata."""
 
-    ATOM = "http://www.w3.org/2005/Atom"
-    OPENSEARCH = "http://a9.com/-/spec/opensearch/1.1/"
-    ARXIV = "http://arxiv.org/schemas/atom"
-    NSMAP = {
-        None: ATOM,
-        "opensearch": OPENSEARCH,
-        "arxiv": ARXIV
-    }
-#     fields = {
-#         'title': '{%s}title' % ATOM,
-#         'id': '{%s}id' % ATOM,
-#         'submitted_date': '{%s}published' % ATOM,
-#         'modified_date': '{%s}updated' % ATOM,
-#         'abstract': '{%s}summary' % ATOM,
-#         ''
-#     }
-#
-#     def __init__(cls, *args, **kwargs) -> None:
-#         super(AtomXMLSerializer, cls).__init__(*args, **kwargs)
-#         cls._root = etree.Element('feed', nsmap=cls.NSMAP)
-#
-#     def transform(cls):
-#         for document in cls.iter_documents():
-#
-#
-#
-#     def __repr__(cls) -> str:
-#         return etree.tostring(cls._root, pretty_print=True)
+    @classmethod
+    def transform_document(cls, fg: FeedGenerator, doc: Document,
+                           query: Optional[APIQuery] = None) -> None:
+        """Select a subset of :class:`Document` properties for public API."""
+        entry = fg.add_entry()
+        entry.id(url_for("api.paper", paper_id=doc['paper_id'],
+                             version=doc['version'], _external=True))
+        entry.title(doc['title'])
+        entry.summary(doc['abstract'])
+        entry.published(doc['submitted_date'])
+        entry.updated(doc['updated_date'])
+        entry.link({'href': url_for("api.paper", paper_id=doc['paper_id'], version=doc['version'], _external=True),
+                    "type": "text/html"})
+        #entry.link({'href': url_for("api.pdf", paper_id=doc['paper_id, version=doc['version, _external=True),
+        #            "type": "application/pdf", 'rel': 'related'})
+
+        if doc['comments']:
+            entry.arxiv.comment(doc['comments'])
+
+        if doc['journal_ref']:
+            entry.arxiv.journal_ref(doc['journal_ref'])
+
+        if doc['doi']:
+            entry.arxiv.doi(doc['doi'])
+        """
+        entry.arxiv.primary_category(doc['primary_classification'].archive['id'])
+        entry.category(
+            term=doc['primary_classification'].archive['id'],
+            scheme='http://arxiv.org://arxiv.org/schemas/atom')
```

Comment:
is that right?

Reply:
Flagged this as a TODO.

Reply:
Isn't it faster to do this
```suggestion
            scheme='http://arxiv.org/schemas/atom')
```
than write a TODO?

Reply:
(or whatever it's supposed to be)

Reply:
Oh, yeah, totally missed that! I flagged it with a TODO because the error is in the middle of a block comment on classifications.